scripts/scripts.js

```diff
@@ -1,617 +1,20 @@
-  const link = document.createElement('link');
-  link.rel = 'icon';
-  link.type = 'image/svg+xml';
-  link.href = href;
-  const existingLink = document.querySelector('head link[rel="icon"]');
-  if (existingLink) {
-    existingLink.parentElement.replaceChild(link, existingLink);
-  } else {
-    document.getElementsByTagName('head')[0].appendChild(link);
-  }
-}
-
-/**
- * load LCP block and/or wait for LCP in default content.
- */
-async function waitForLCP() {
-  // eslint-disable-next-line no-use-before-define
-  const lcpBlocks = LCP_BLOCKS;
-  const block = document.querySelector('.block');
-  const hasLCPBlock = (block && lcpBlocks.includes(block.getAttribute('data-block-name')));
-  if (hasLCPBlock) await loadBlock(block, true);
-
-  document.querySelector('body').classList.add('appear');
-  const lcpCandidate = document.querySelector('main img');
-  await new Promise((resolve) => {
-    if (lcpCandidate && !lcpCandidate.complete) {
-      lcpCandidate.setAttribute('loading', 'eager');
-      lcpCandidate.addEventListener('load', () => resolve());
-      lcpCandidate.addEventListener('error', () => resolve());
-    } else {
-      resolve();
-    }
-  });
-}
-
-/**
- * Decorates the page.
- */
-async function loadPage(doc) {
-  // eslint-disable-next-line no-use-before-define
-  await loadEager(doc);
-  // eslint-disable-next-line no-use-before-define
-  await loadLazy(doc);
-  // eslint-disable-next-line no-use-before-define
-  loadDelayed(doc);
-}
-
-export function initHlx() {
-  window.hlx = window.hlx || {};
-  window.hlx.lighthouse = new URLSearchParams(window.location.search).get('lighthouse') === 'on';
-  window.hlx.codeBasePath = '';
-
-  const scriptEl = document.querySelector('script[src$="/scripts/scripts.js"]');
-  if (scriptEl) {
-    try {
-      [window.hlx.codeBasePath] = new URL(scriptEl.src).pathname.split('/scripts/scripts.js');
-    } catch (error) {
-      // eslint-disable-next-line no-console
-      console.log(error);
-    }
-  }
-}
-
-initHlx();
-
-/*
- * ------------------------------------------------------------
- * Edit above at your own risk
- * ------------------------------------------------------------
- */
-
-const LCP_BLOCKS = ['hero']; // add your LCP blocks to the list
-const RUM_GENERATION = 'project-1'; // add your RUM generation information here
-const ICON_ROOT = '/icons';
-
-sampleRUM('top');
-
-window.addEventListener('load', () => sampleRUM('load'));
-
-window.addEventListener('unhandledrejection', (event) => {
-  sampleRUM('error', { source: event.reason.sourceURL, target: event.reason.line });
-});
-
-window.addEventListener('error', (event) => {
-  sampleRUM('error', { source: event.filename, target: event.lineno });
-});
-
-loadPage(document);
+import {
+  sampleRUM,
+  buildBlock,
+  loadHeader,
+  loadFooter,
+  decorateButtons,
+  decorateIcons,
+  decorateSections,
+  decorateBlocks,
+  decorateTemplateAndTheme,
+  waitForLCP,
+  loadBlocks,
+  loadCSS,
+} from './block-utils.js';
```

Comment:
I agree, especially since there is stuff in there that is not specific to blocks. like the `sampleRum`

Reply:
Why not just call it `utils`?

Reply:
My feeling is a name like `utils` would invite people to add their own code in there which I'm assuming we don't want them to do. What about `franklin-utils` or `franklin-lib`. Or do we not want to tie ourselves too closely to the franklin name like we did with `helix`?

Reply:
> My feeling is a name like utils would invite people to add their own code in there which I'm assuming we don't want them to do. 

Right... but wouldn't the same be true for `block-utils`?

> What about `franklin-utils` or `franklin-lib`. Or do we not want to tie ourselves too closely to the franklin name like we did with `helix`?

I'm guessing that was exactly @davidnuescheler's thinking behind the `block-utils.js` name... what I don't like about it is that it implies that the helpers in it are only meant for use inside blocks, when in fact many of them are meant to be used in `scripts.js`...

I kind of like `lib.js` as a deterrent against project specific additions. My other suggestions would be `boilerplate.js` or `core.js`.

Reply:
i agree that we should carefully name this lib as it will be around for a while and changes to the name would be relevant for all imports...

here is my thinking, and i am open to discussing this... 
i didn't want to use `franklin` as name as i do have a bit of a hangover from the last project name change.
i am not really worried that there are things in there that are not strictly `block` only related, and to me it sounded intuitive as the majority of the surface that a dev uses is `block` related.  so in a way i applied the `intuitive, simple over consistent` rule... https://github.com/adobe/helix-home/blob/main/manifesto.md#intuitive-simple--consistent-p

i also, would find `lib-franklin.js` the most intuitive name and i think there is some element of "product" that i find appealing.